core/src/main/java/org/elasticsearch/search/controller/SearchPhaseController.java

```diff
@@ -258,40 +274,79 @@ private static long optionalSum(long left, long right) {
             }
             mergedTopDocs = TopDocs.merge(from, topN, shardTopDocs);
         }
-        return mergedTopDocs.scoreDocs;
-    }
 
-    public ScoreDoc[] getLastEmittedDocPerShard(SearchRequest request, ScoreDoc[] sortedShardList, int numShards) {
-        if (request.scroll() != null) {
-            return getLastEmittedDocPerShard(sortedShardList, numShards);
-        } else {
-            return null;
+        ScoreDoc[] scoreDocs = mergedTopDocs.scoreDocs;
```

Comment:
as far as I can tell this is only relevant in a scroll context. I am not 100% following why you implemented this here?